src/server/scripts/Outland/BlackTemple/boss_illidan.cpp

```diff
@@ -50,2180 +37,2571 @@ enum Say
-                            Glaive->SetDisplayId(MODEL_INVISIBLE);
-                            Glaive->setFaction(me->getFaction());
-                            DoCast(Glaive, SPELL_THROW_GLAIVE, true);
-                        }
+                        float angle = me->GetAngle(IllidanMiddlePoint);
+                        me->SetFacingTo(angle, true);
+                        break;
                     }
-                    Timer[EVENT_FLIGHT_SEQUENCE] = 5000;
-                    break;
-                case 5: // summon flames
-                    SummonFlamesOfAzzinoth();
-                    Timer[EVENT_FLIGHT_SEQUENCE] = 3000;
-                    break;
-                case 6: // fly to hover point
-                    me->GetMotionMaster()->MovePoint(0, HoverPosition[HoverPoint].x, HoverPosition[HoverPoint].y, HoverPosition[HoverPoint].z);
-                    Timer[EVENT_FLIGHT_SEQUENCE] = 0;
-                    break;
-                case 7: // return to center
-                    me->GetMotionMaster()->MovePoint(0, CENTER_X, CENTER_Y, CENTER_Z);
-                    Timer[EVENT_FLIGHT_SEQUENCE] = 0;
-                    break;
-                case 8: // glaive return
-                    for (uint8 i = 0; i < 2; ++i)
+                    case EVENT_EYE_BLAST:
                     {
-                        if (GlaiveGUID[i])
+                        events.CancelEvent(EVENT_DARK_BARRAGE);
+                        Position pos = IllidanDBTargetSpawnPositions[_dbTargetIndex];
+                        if (TempSummon* dbTarget = me->SummonCreature(NPC_ILLIDAN_DB_TARGET, pos, TEMPSUMMON_MANUAL_DESPAWN))
                         {
-                            Unit* Glaive = ObjectAccessor::GetUnit(*me, GlaiveGUID[i]);
-                            if (Glaive)
-                            {
-                                Glaive->CastSpell(me, SPELL_GLAIVE_RETURNS, false); // Make it look like the Glaive flies back up to us
-                                Glaive->SetDisplayId(MODEL_INVISIBLE); // disappear but not die for now
-                            }
+                            Talk(SAY_ILLIDAN_EYE_BLAST);
+                            DoCast(dbTarget, SPELL_EYE_BLAST);
+                            dbTarget->GetMotionMaster()->MovePoint(POINT_DB_TARGET, IllidanDBTargetPoints[_dbTargetIndex]);
                         }
+                        break;
                     }
-                    Timer[EVENT_FLIGHT_SEQUENCE] = 2000;
-                    break;
-                case 9: // land
-                    me->SetDisableGravity(false);
-                    me->StopMoving();
-                    me->HandleEmoteCommand(EMOTE_ONESHOT_LAND);
-                    for (uint8 i = 0; i < 2; ++i)
+                    case EVENT_DARK_BARRAGE:
                     {
-                        if (GlaiveGUID[i])
-                        {
-                       
```

Comment:
Ah yes, can be changed. I did in c++ to use PassiveAI, and i hate do stuffs in SAI/DB y.y